client.go

```diff
@@ -250,7 +266,43 @@ func (c *Client) send(ctx context.Context, state *lib.State, buf []byte, useProt
 	return *response, err
 }
 
-func ResponseCallback(n int) bool {
+func IsSuccessfulResponse(n int) bool {
 	// report all 2xx respones as successful requests
 	return n/100 == 2
 }
+
+type responseWithStats struct {
+	Data struct {
+		Stats stats.Result
+	}
+}
+
+func reportMetricsFromStats(ctx context.Context, response httpext.Response, queryType QueryType) error {
+	responseBody, ok := response.Body.(string)
+	if !ok {
+		return errors.New("response body is not a string")
+	}
+	responseWithStats := responseWithStats{}
+	err := json.Unmarshal([]byte(responseBody), &responseWithStats)
+	if err != nil {
+		return errors.Wrap(err, "error unmarshalling response body to response with stats")
+	}
+	now := time.Now()
+	k6_stats.PushIfNotDone(ctx, lib.GetState(ctx).Samples, k6_stats.ConnectedSamples{
+		Samples: []k6_stats.Sample{
+			{
+				Metric: BytesProcessedPerSecondsSummary,
+				Tags:   k6_stats.NewSampleTags(map[string]string{"endpoint": queryType.Endpoint()}),
+				Value:  float64(responseWithStats.Data.Stats.Summary.BytesProcessedPerSecond),
+				Time:   now,
+			},
+			{
+				Metric: LinesProcessedPerSecondsSummary,
+				Tags:   k6_stats.NewSampleTags(map[string]string{"method": queryType.Endpoint()}),
```

Comment:
We should use the same tags in both metrics.

```suggestion
				Tags:   k6_stats.NewSampleTags(map[string]string{"endpoint": queryType.Endpoint()}),
```

Reply:
done